lib/Dialect/FIRRTL/PassDetails.h

```diff
@@ -0,0 +1,20 @@
+//===- PassDetails.h - FIRRTL pass class details ----------------*- C++ -*-===//
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef DIALECT_FIRRTL_PASSDETAILS_H
+#define DIALECT_FIRRTL_PASSDETAILS_H
+
+#include "circt/Dialect/FIRRTL/Ops.h"
+#include "mlir/Pass/Pass.h"
+
+namespace circt {
+namespace firrtl {
+
+#define GEN_PASS_CLASSES
```

Comment:
I copied this pattern from LLHD. Clang tidy is taking offense to the `#ifndef` guards... I will take a look at that.

Reply:
IIRC clang-tidy doesn't like headers outside of include dirs.

Reply:
This include is needed in both LowerToRTL and LowerTypes, and I thought just putting the include in both .cpp files was causing issues. That may have just been something else I got wrong when I was first setting up the pass... I will check again. I saw LLHD does it this way with a [header](https://github.com/llvm/circt/blob/master/lib/Dialect/LLHD/Transforms/PassDetails.h) in the lib, but if it isn't needed I'll remove it.

Reply:
Ok, double checked and I will just remove this file and have the include in both .cpp files.

Reply:
headers in lib directories is a good thing.  This is a "private header" intended to share code between .cpp files in a library, but which isnt' itself a public interface.  If you need to share code between cpp files, then this is exactly the right thing, thanks!

Reply:
Ok, keeping this "private header" to share the generated pass implementations between whatever files define passes.